compiler/rustc_middle/src/query/mod.rs

```diff
@@ -2044,4 +2044,10 @@ rustc_queries! {
         desc { |tcx| "looking up generator diagnostic data of `{}`", tcx.def_path_str(key) }
         separate_provide_extern
     }
+
+    query compare_assoc_const_impl_item_with_trait_item(
+        key: (&'tcx ty::AssocItem, &'tcx ty::AssocItem, ty::TraitRef<'tcx>)
```

Comment:
Can the query directly take 3 `DefId`s and call `associated_item` and `impl_trait_ref` on its own?

Reply:
ah yea it could probably even take `(LocalDefId, DefId)`